frontend/src/reducers/cartReducers.js

```diff
@@ -0,0 +1,37 @@
+import { ADD_TO_CART, REMOVE_ITEM_CART } from "../constants/cartConstants";
+
+export const cartReducer = (state = { cartItems: [] }, action) => {
+  switch (action.type) {
+    case ADD_TO_CART:
+      const item = action.payload;
+
+      const isItemExist = state.cartItems.find(
+        (i) => i.product === item.product
+      );
+
+      if (isItemExist) {
+        return {
+          ...state,
+          cartItems: state.cartItems.map((i) =>
```

Comment:
suggestion: use more descriptive var instead of using ( i ) here

Reply:
I will keep in mind.